src/lib/provable/test/ecdsa.unit-test.ts

```diff
@@ -189,3 +190,36 @@ console.timeEnd('ecdsa verify (prove)');
 
 assert(await program.verify(proof), 'proof verifies');
 proof.publicOutput.assertTrue('signature verifies');
+
+// check constraints w/o endomorphism
+
+let programNoEndo = ZkProgram({
+  name: 'ecdsa-secp256r1',
+  publicOutput: Bool,
+  methods: {
+    ecdsa: {
+      privateInputs: [],
+      async method() {
+        let signature_ = Provable.witness(Ecdsa.Signature, () => signature);
+        let msgHash_ = Provable.witness(Field3, () => msgHash);
+        let publicKey_ = Provable.witness(Point, () => publicKey);
+
+        return {
+          publicOutput: Ecdsa.verify(
+            Secp256r1,
+            signature_,
+            msgHash_,
+            publicKey_,
+            config
+          ),
+        };
+      },
+    },
+  },
+});
+
+console.time('ecdsa verify, no endomorphism (build constraint system)');
+let csNoEndo = (await programNoEndo.analyzeMethods()).ecdsa;
```

Comment:
Could we add some test vectors to verify this implementation?

Reply:
yes! I'll do that